med-assist-backend/src/main/java/inc/evil/medassist/appointment/facade/AppointmentFacadeImpl.java

```diff
@@ -71,15 +66,31 @@ public AppointmentResponse update(String id, UpsertAppointmentRequest request) {
         return AppointmentResponse.from(appointmentService.update(id, originalAppointment.mergeWith(newAppointment)));
     }
 
-    private Appointment toAppointment(UpsertAppointmentRequest request) {
-        return Appointment.builder()
-                .appointmentDate(request.getAppointmentDate())
-                .startTime(request.getStartTime())
-                .endTime(request.getEndTime())
+    @Override
+    @Transactional
+    public AppointmentResponse create(UpsertAppointmentRequest request) {
+        Appointment appointmentToCreate = toAppointment(request);
+        Appointment createdAppointment = appointmentService.create(appointmentToCreate);
+        return AppointmentResponse.from(createdAppointment);
+    }
+
+    private Appointment toAppointment(final UpsertAppointmentRequest request) {
+        Appointment appointmentToCreate = Appointment.builder()
+                .appointmentDate(request.getStartDate() != null ? request.getStartDate().toLocalDate() : null)
+                .startTime(request.getStartDate() != null ? request.getStartDate().toLocalTime() : null)
+                .endTime(request.getEndDate() != null ? request.getEndDate().toLocalTime() : null)
                 .operation(request.getOperation())
                 .doctor(request.getDoctorId() != null ? doctorService.findById(request.getDoctorId()) : null)
-                .patient(request.getPatientId() != null ? patientService.findById(request.getPatientId()) : null)
```

Comment:
Why don't we add a method like this:
```java
private Patient findOrCreatePatient(final UpsertAppointmentRequest request) {
        if(request.isExistingPatient()){
           return request.getPatientId() != null ? patientService.findById(request.getPatientId()) : null;
        } else {
           return patientService.create(request.getPatientRequest().toPatient());
        }
}
```

and then while building the `appointmentToCreate` object we can have something like:

```java
...
.doctor(request.getDoctorId() != null ? doctorService.findById(request.getDoctorId()) : null)
.patient(findOrCreatePatient(request))
...
```

Reply:
agree, great idea